### PR TITLE
Model validations and Testing

### PR DESCRIPTION
**Pull request by marvin-nyalik**
In this branch:

- I added the following validations:
User model:
  *Name must not be blank.
  *PostsCounter must be an integer greater than or equal to zero.
Post model:
  *Title must not be blank.
  *Title must not exceed 250 characters.
  *CommentsCounter must be an integer greater than or equal to zero.
  *LikesCounter must be an integer greater than or equal to zero.

-I added unit specs for the user and post methods and validations.

Let me know if changes are needed @marvin-nyalik

### CODE REVIEW

File: spec/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  it 'name should be present' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'postsCounter should allow valid values' do
+    subject.postsCounter = -2
+    expect(subject).to_not be_valid
+  end
```
Comment (TadesseDev) [inline spec/user_spec.rb:16]: - [ ] Dear @marvin-nyalik, good job adding the `user_spec` test file to test your user model, but we need to keep in mind that any custom methods in the models should have a corresponding unit test. and you are missing to test the `recent_posts` method which is inside the `use` model.

File: spec/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  subject { Post.new(title: 'My post', text: 'My post body', author_id: 1, commentsCounter: 12, likesCounter: 3) }
+
+  before { subject.save }
+
+  it 'title should be present' do
+    subject.title = nil
```
Comment (TadesseDev) [inline spec/post_spec.rb]: - [ ] As mentioned earlier, to make sure any custom methods in the models have a corresponding unit test, Please add a test that will assert the `comments_for_post` and `update_user_posts_counter` methods, which are inside the `post` model.

File: spec/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
```
Comment (TadesseDev) [inline spec/user_spec.rb:2]: - [ ] Also, As stated in the requirements, we need to add unit tests for the `User`, `Post`, `Comment`, and `Like` models, currently, I could not find a test/spec file for the  `Comment`, and `Like` models. Please add this file and make sure to test for `validation` alongside other custom methods inside those models.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @marvin-nyalik 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] You have written a clean code ✅
 - [x]  Most of the requirement has been meet. ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/user_spec.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply spec/user_spec.rb:16]: Fixed.

File: spec/post_spec.rb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply spec/post_spec.rb]: Fixed.

File: spec/user_spec.rb (hunk 3 above)
Comment (Abenezer-Tilahun) [reply spec/user_spec.rb:2]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @marvin-nyalik 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.
✔️ All Test are passing.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**